### PR TITLE
adding some type safe handling for some of the user value accessors and some more user values access batchings

### PR DESCRIPTION
**Pull request by eishay**
@ymatsuda @andrewconner @2is10

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/common/social/UserConnectionCreator.scala [hunk 1]
```diff
@@ -48,7 +48,7 @@ class UserConnectionCreator @Inject() (
   }
 
   def getConnectionsLastUpdated(userId: Id[User]): Option[DateTime] = db.readOnly { implicit s =>
-    userValueRepo.getValue(userId, UserConnectionCreator.UpdatedUserConnectionsKey) map parseStandardTime
+    userValueRepo.getValueUnsafe(userId, UserConnectionCreator.UpdatedUserConnectionsKey) map parseStandardTime
```
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/common/social/UserConnectionCreator.scala]: How about renaming `getValueUnsafe` → `getValueStringOpt`. The word “unsafe” makes it sound like one should feel guilty for using it.

File: kifi-backend/modules/common/app/com/keepit/model/UserValue.scala [hunk 2]
```diff
@@ -44,3 +45,36 @@ object Gender {
...
+  val showSearchIntro = UserValueBooleanHandler("ext_show_search_intro", false)
+  val showFindFriends = UserValueBooleanHandler("ext_show_find_friends", false)
+
+  val UserInitPrefs: Seq[String] = Seq(enterToSend, maxResults, showKeeperIntro, showSearchIntro, showFindFriends).map(_.name)
+
+  val availableInvites = UserValueIntHandler("availableInvites", 1000)
+  val hasSeenInstall = UserValueBooleanHandler("has_seen_install", false)
+  val welcomeEmailSent = UserValueBooleanHandler("welcomeEmailSent", false)
+
+}
```
Comment (2is10) [inline kifi-backend/modules/common/app/com/keepit/model/UserValue.scala]: The red indicator here means the file ends without the conventional terminating `\n`.

**Comment by 2is10** [comment]
looks good to me. seeing:

> We can’t automatically merge this pull request.

File: kifi-backend/modules/shoebox/app/com/keepit/common/social/UserConnectionCreator.scala (hunk 1 above)
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/common/social/UserConnectionCreator.scala]: Its a bit unsafe in the sense that its not type safe, but I agree, I'll rename

Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/common/social/UserConnectionCreator.scala]: thanks. i think the longer name is penalty enough :smile:

**Comment by eishay** [comment]
updated

File: tools/etc/nginx/conf.d/shoebox.conf [hunk 3]
```diff
@@ -10,6 +10,10 @@ upstream shoebox {
     server localhost:9000;
 }
 
+upstream seyren {
+    server 50.18.107.175:8080;
```
Comment (2is10) [inline tools/etc/nginx/conf.d/shoebox.conf:14]: did you mean for this to be part of the pull request?

**Pull request merged**